AmplifyPlugins/Core/AWSPluginsCoreTests/Model/Decorator/AuthRuleDecorator/ModelWithOwnerAuthAndMultiGroup.swift

```diff
@@ -0,0 +1,242 @@
+        ]
+        model.pluralName = "OIDCMultiGroupPosts"
+        model.fields(
+            .id(),
+            .field(oIDCMultiGroupPost.title, is: .required, ofType: .string),
+            .field(oIDCMultiGroupPost.owner, is: .optional, ofType: .string)
+        )
+    }
+}
+
+class ModelWithOwnerAuthAndMultiGroup: XCTestCase {
+    override func setUp() {
+        ModelRegistry.register(modelType: OIDCMultiGroupPost.self)
+    }
+
+    override func tearDown() {
+        ModelRegistry.reset()
+    }
+
+    func testOnCreateSubscription_NoGroupInfoPassed() {
+        let claims = ["username": "user1",
+                      "sub": "123e4567-dead-beef-a456-426614174000"] as IdentityClaimsDictionary
+        var documentBuilder = ModelBasedGraphQLDocumentBuilder(modelType: OIDCMultiGroupPost.self,
+                                                               operationType: .subscription)
+        documentBuilder.add(decorator: DirectiveNameDecorator(type: .onCreate))
+        documentBuilder.add(decorator: AuthRuleDecorator(.subscription(.onCreate, claims)))
+        let document = documentBuilder.build()
+        let expectedQueryDocument = """
+        subscription OnCreateOIDCMultiGroupPost($owner: String!) {
+          onCreateOIDCMultiGroupPost(owner: $owner) {
+            id
+            owner
+            title
+            __typename
+          }
+        }
+        """
+        XCTAssertEqual(document.name, "onCreateOIDCMultiGroupPost")
+        XCTAssertEqual(document.stringValue, expectedQueryDocument)
+        guard let variables = document.variables else {
+            XCTFail("The document doesn't contain variables")
+            return
+        }
+        XCTAssertEqual(variables["owner"] as? String, "123e4567-dead-beef-a456-426614174000")
+    }
+
+    func testOnCreateSubscription_InDifferentAppWithModerators() {
+        let claims = ["username": "user1",
+                      "sub": "123e4567-dead-beef-a456-426614174000",
+                      "https://differentapp.com/claims/groups": ["Moderators"]] as IdentityClaimsDictionary
+        var documentBuilder = ModelBasedGraphQLDocumentBuilder(modelType: OIDCMultiGroupPost.self,
+                                                               operationType: .subscription)
+        documentBuilder.add(decorator: DirectiveNameDecorator(type: .onCreate))
+        documentBuilder.add(decorator: AuthRuleDecorator(.subscription(.onCreate, claims)))
+        let document = documentBuilder.build()
+        let expectedQueryDocument = """
+        subscription OnCreateOIDCMultiGroupPost {
+          onCreateOIDCMultiGroupPost {
+            id
+            owner
+            title
+            __typename
+          }
+        }
+        """
+        XCTAssertEqual(document.name, "onCreateOIDCMultiGroupPost")
+        XCTAssertEqual(document.stringValue, expectedQueryDocument)
+        XCTAssertTrue(document.variables.isEmpty)
```

Comment:
```suggestion
        XCTAssertTrue(document.variables.isEmpty, "variables should be empty since claim group value matches the schema")
```

Reply:
Chagned to:
"variables should be empty since claim group value matches the auth rule schema"